client/src/app/pages/waves/waves.tsx

```diff
@@ -0,0 +1,330 @@
+import { useSortState } from "@app/shared/hooks/useSortState";
+import { Application, Wave } from "@app/api/models";
+import {
+  FilterCategory,
+  FilterToolbar,
+  FilterType,
+} from "@app/shared/components/FilterToolbar";
+import { useFilterState } from "@app/shared/hooks/useFilterState";
+import { useSelectionState } from "@migtools/lib-ui";
+import { TdProps } from "@patternfly/react-table";
+import dayjs from "dayjs";
+import { IComposableRow } from "@app/shared/components/composable-app-table/composable-app-table";
+import { WaveApplicationsTable } from "./wave-applications-table/wave-applications-table";
+import { WaveStakeholdersTable } from "./wave-stakeholders-table/wave-stakeholders-table";
+
+export const Waves: React.FC = () => {
+  const { t } = useTranslation();
+
+  const { waves, isFetching, fetchError } = useFetchWaves();
+
+  const filterCategories: FilterCategory<Wave>[] = [
+    {
+      key: "name",
+      title: t("terms.name"),
+      type: FilterType.search,
+      placeholderText:
+        t("actions.filterBy", {
+          what: t("terms.name").toLowerCase(),
+        }) + "...",
+      getItemValue: (item) => {
+        return item?.name || "";
+      },
+    },
+  ];
+
+  const handleOnClearAllFilters = () => {
+    setFilterValues({});
+  };
+
+  const { filterValues, setFilterValues, filteredItems } = useFilterState(
+    waves || [],
+    filterCategories
+  );
+  //Bulk selection
+  const {
+    isItemSelected: isRowSelected,
+    toggleItemSelected: toggleRowSelected,
+    selectAll,
+    selectMultiple,
+    areAllSelected,
+    selectedItems: selectedRows,
+  } = useSelectionState<Wave>({
+    items: filteredItems || [],
+    isEqual: (a, b) => a.id === b.id,
+  });
+
+  const getSortValues = (item: Wave) => [
+    "",
+    item?.name || "",
+    "", // Action column
+  ];
+
+  const { sortBy, onSort, sortedItems } = useSortState(
+    filteredItems,
+    getSortValues
+  );
+
+  const { currentPageItems, setPageNumber, paginationProps } =
+    usePaginationState(sortedItems, 10);
+
+  const columnNames = {
+    select: "",
+    name: "Name",
+    startDate: "Start date",
+    endDate: "End date",
+    applications: "Applications",
+    stakeholders: "Stakeholders",
+    status: "Status",
+    actions: "",
+  };
+
+  //Compound expandable code
+  type ColumnKey = keyof typeof columnNames;
+
+  const [expandedCells, setExpandedCells] = React.useState<
+    Record<string, ColumnKey>
+  >({
+    wave1: "applications", // Default to the first cell of the first row being expanded
+  });
+  const setCellExpanded = (
+    application: Application,
+    columnKey: ColumnKey,
+    isExpanding = true
+  ) => {
+    const newExpandedCells = { ...expandedCells };
+    if (isExpanding) {
+      newExpandedCells[application.name] = columnKey;
+    } else {
+      delete newExpandedCells[application.name];
+    }
+    setExpandedCells(newExpandedCells);
+  };
+  const compoundExpandParams = (
```

Comment:
![image](https://user-images.githubusercontent.com/11218376/224819123-e0eca917-dc4a-41b2-80b3-0b4e04471297.png)

The expandedCells object is a dictionary of {applicationName: 'stakeholder's | 'applications'} values to handle the expanded state